core/src/main/java/org/moditect/internal/compiler/ModuleInfoCompiler.java

```diff
@@ -135,12 +134,12 @@ public static ModuleDeclaration parseModuleInfo(String moduleInfoSource) {
         return classWriter.toByteArray();
     }
 
-    private static String getNameForBinary(Type type) {
-        return getNameForBinary( type.toString() );
+    private static String getNameForBinary(Name name) {
+        return getNameForBinary( name.asString() );
```

Comment:
Better not use "toString" since that also outputs comments and annotations and so on.

Reply:
Good catch!